experimental/devices/ads1x15/ads1x15.go

```diff
@@ -0,0 +1,392 @@
+package ads1x15
+
+import (
+	"encoding/binary"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/pin"
+
+	"fmt"
+
+	"periph.io/x/periph/conn/i2c"
+)
+
+const (
+
+	// ADS1x15DefaultAddress is the default I2C address for the ADS1x15 components
+	ADS1x15DefaultAddress = 0x48
+
+	ads1x15PointerConversion    = 0x00
+	ads1x15PointerConfig        = 0x01
+	ads1x15PointerLowThreshold  = 0x02
+	ads1x15PointerHighThreshold = 0x03
+	// Write: Set to start a single-conversion
+	ads1x15ConfigOsSingle       = 0x8000
+	ads1x15ConfigMuxOffset      = 12
+	ads1x15ConfigModeContinuous = 0x0000
+	//Single shoot mode
+	ads1x15ConfigModeSingle = 0x0100
+
+	ads1x15ConfigCompWindow      = 0x0010
+	ads1x15ConfigCompAactiveHigh = 0x0008
+	ads1x15ConfigCompLatching    = 0x0004
+	ads1x15ConfigCompQueDisable  = 0x0003
+
+	Channel0 = 0
+	Channel1 = 1
+	Channel2 = 2
+	Channel3 = 3
+)
+
+// Dev is the driver for the ADS1015/ADS1115 ADC
+type Dev struct {
+	// I2C Communication
+	c i2c.Dev
+
+	name string
+
+	gainConfig      map[int]uint16
+	dataRates       map[int]uint16
+	gainVoltage     map[int]physic.ElectricPotential
+	DefaultDataRate int
+	mutex           *sync.Mutex
+}
+
+// AnalogPin represents a pin which is able to read an electric potential
+type AnalogPin interface {
+	pin.Pin
+	Read() (physic.ElectricPotential, error)
+}
+
+type ads1x15AnalogPin struct {
```

Comment:
I'd like the 4 pins to be stored inside Dev as:

```
type Dev struct {
   ...
   pins [4] pin
}
```

No need to specify the whole " ads1x15Analog" prefix IMHO.